### PR TITLE
[fix](select join) Make selected slotRef nullable when slotRef is from nullable tuple in outer join sql block

### LINKED ISSUE #7288: [Bug] Execute left out join sql with group_concat function will cause be core
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-doris/issues?q=is%3Aissue) and found no similar issues.

### Version

master

### What's Wrong?

outer join sql would make some slot nullable when slot is selected from nullable tuple in outer join sql even the origin slot is not nullable. but now our design forget to check it. 

### What You Expected?

be not core

### How to Reproduce?

 CREATE TABLE `left_table` (
  `k1` int(11) NULL COMMENT "",
  `no` varchar(50) NOT NULL COMMENT ""
) ENGINE=OLAP
DUPLICATE KEY(`k1`)
COMMENT "OLAP"
DISTRIBUTED BY HASH(`k1`) BUCKETS 2
PROPERTIES (
"replication_allocation" = "tag.location.default: 1",
"in_memory" = "false",
"storage_format" = "V2"
);

CREATE TABLE `right_table` (
  `k1` int(11) NULL COMMENT "",
  `no` varchar(50) NOT NULL COMMENT ""
) ENGINE=OLAP
DUPLICATE KEY(`k1`)
COMMENT "OLAP"
DISTRIBUTED BY HASH(`k1`) BUCKETS 2
PROPERTIES (
"replication_allocation" = "tag.location.default: 1",
"in_memory" = "false",
"storage_format" = "V2"
) ;
insert into left_table values(1, "test");
select l.k1, group_concat(r.no) from left_table l left join right_table r on l.k1=r.k1 group by l.k1; 

### Anything Else?

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by caiconghui**
## Proposed changes

Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. If it fixes a bug or resolves a feature request, be sure to link to that issue.

## Types of changes

What types of changes does your code introduce to Doris?
_Put an `x` in the boxes that apply_

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Code refactor (Modify the code structure, format the code, etc...)
- [ ] Optimization. Including functional usability improvements and performance improvements.
- [ ] Dependency. Such as changes related to third-party components.
- [ ] Other.

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I have created an issue on (Fix #7288) and described the bug/feature there in detail
- [ ] Compiling and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If these changes need document changes, I have updated the document
- [ ] Any dependent changes have been merged

## Further comments

If this is a relatively large or complex change, kick off the discussion at dev@doris.apache.org by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Review by morningman — COMMENTED** [review]
LGTM cc @EmmyMiao87

File: be/src/exprs/aggregate_functions.cpp [hunk 1]
```diff
@@ -838,7 +838,6 @@ void AggregateFunctions::string_concat_update(FunctionContext* ctx, const String
     if (result->is_null || !result->ptr) {
         // Header of the intermediate state holds the length of the first separator.
         const auto header_len = sizeof(StringConcatHeader);
-        DCHECK(header_len == sizeof(sep->len));
```
Comment (EmmyMiao87) [inline be/src/exprs/aggregate_functions.cpp:841]: Why was this logic removed?

Comment (caiconghui) [reply be/src/exprs/aggregate_functions.cpp:841]: header_len is const value now, but sep len can be various

**Review by EmmyMiao87 — APPROVED** [review]
LGTM

**Pull request merged**